test/widgets/test_mpris2widget.py

```diff
@@ -0,0 +1,264 @@
+# Copyright (c) 2021 elParaguayo
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# Widget specific tests
+
+# The test_widget_init_config will cover the scenario where `kbdd` is not
+# running and will also run the asyncio _config_async method.
+
+# This test file covers the remaining widget code
+
+import sys
+from importlib import reload
+from types import ModuleType
+
+import pytest
+
+from libqtile.bar import Bar
+
+
+def no_op(*args, **kwargs):
+    pass
+
+
+def fake_timer(interval, func, *args, **kwargs):
+    class TimerObj:
+        def cancel(self):
+            pass
+
+    return TimerObj()
+
+
+class MockConstants(ModuleType):
+    class MessageType:
+        SIGNAL = 1
+
+
+class MockMessage:
+    def __init__(self, is_signal=True, body=None):
+        self.message_type = 1 if is_signal else 0
+        self.body = body
+
+
+class obj:  # noqa: N801
+    def __init__(self, value):
+        self.value = value
```

Comment:
`dbus_next` message data is stored in a `Variant` object. The widget extracts the information from the `value` attribute so that's all we need to mock here.